### PR TITLE
Remove cache file support

### PR DESCRIPTION
**Pull request by jukkar**
As we can get the full state of the pod and container from runtime, no need to cache the data.
Containerd already has support for this. For cri-o, one needs a version that has commit f59c1f72a837c2235010813b8f361bb57c8c551b ("sandbox: Handle PodLinuxOverhead and PodLinuxResources CRI fields") in it.

### CODE REVIEW

File: pkg/resmgr/cache/cache.go [hunk 1]
```diff
@@ -1027,6 +998,10 @@ func (cch *cache) mkdirAll(what, path string, p *permissions) error {
 	return nil
 }
 
+func (cch *cache) SerializeAsJSON() ([]byte, error) {
```
Comment (klihub) [inline pkg/resmgr/cache/cache.go:1001]: nitpick: should/could we use `cache.Snapshot()` here directly ? It is already exported. Or rename it if we don't like the name anymore ?

Also, if we've now nuked all code paths that might externally lock the cache and then invoke (indirectly) `cache.Snapshot()` (via `cache.Save()`), then I think we should lock the cache now while we take the snapshot. The command line option is always present. When enabled this function can be triggered externally. And while taking the snapshot, we iterate over the container and pod maps.

File: pkg/resmgr/flags.go [hunk 2]
```diff
@@ -66,6 +67,7 @@ func init() {
 		"Configuration used to override the one stored in the cache. Disables the agent.")
 	flag.StringVar(&opt.ForceConfigSignal, "force-config-signal", "SIGHUP",
 		"Signal used to reload forced configuration.")
+	flag.BoolVar(&opt.CacheOutputEnable, "cache-output-enable", false, "Support cache output via HTTP handler.")
```
Comment (klihub) [inline pkg/resmgr/flags.go:70]: nitpick: Could we find/use a more descriptive command line option name for this ? Maybe `--enable-cache-http-query`, but I'm not good with names, so this is just a vague attempt at a name.

File: test/e2e/files/nri-resource-policy-balloons-deployment.yaml.in [hunk 3]
```diff
@@ -66,6 +66,7 @@ spec:
             - /tmp/nri-resource-policy.pid
             - -metrics-interval
             - 5s
+            - --cache-output-enable
```
Comment (klihub) [inline test/e2e/files/nri-resource-policy-balloons-deployment.yaml.in:69]: nitpick: same nitpick, this sort of sticks out as 'I can't guess what this enables'...

**Review by klihub — CHANGES_REQUESTED** [review]
LGTM. I have a few nitpicks and a locking-related comment.

**Review by klihub — COMMENTED** [review]
Looks go to me now. Maybe we'll need to sit on this for a while, at least until the first CRI-O release is done with the merged related fixes.

And maybe we'll need to add a version check to pkg/resmgr/nri.go:Configure() to give an error (and maybe refuse to start up) if the runtime+version indicates a cri-o without the fixes.

**Comment by jukkar** [comment]
Fixed the format error. We can address the cri-o version check later after the cri-o release.

**Comment by jukkar** [comment]
Removed the cache rm task from todo file.

**Comment by jukkar** [comment]
Rebased to main. Changed option name used by e2e test to `--enable-test-apis` to sync it with the e2e testing option introduced in #60

**Comment by klihub** [comment]
We'll revisit this later. Closing for now.

**Pull request closed without merging**